src/libstd/sys_common/util.rs

```diff
@@ -34,4 +37,21 @@ pub fn abort(args: fmt::Arguments) -> ! {
 pub unsafe fn report_overflow() {
     dumb_print(format_args!("\nthread '{}' has overflowed its stack\n",
                             thread::current().name().unwrap_or("<unknown>")));
+
+    #[cfg(feature = "backtrace")]
+    {
+        let log_backtrace = backtrace::log_enabled();
+
+        use sync::atomic::{AtomicBool, Ordering};
+
+        static FIRST_OVERFLOW: AtomicBool = AtomicBool::new(true);
+
+        if let Some(format) = log_backtrace {
+            if let Ok(mut stderr) = Stderr::new() {
+                let _ = backtrace::print(&mut stderr, format);
```

Comment:
Could you please print `thread '{}' has overflowed its stack` here too, because the backtrace is likely to be very big, which makes it easy to miss the first time it is printed.

Reply:
Excellent idea.